### PR TITLE
Validate creating descriptor pool with mutable types

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

File: layers/parameter_validation_utils.cpp [hunk 1]
```diff
@@ -4680,6 +4680,66 @@ bool StatelessValidation::manual_PreCallValidateCreateWin32SurfaceKHR(VkInstance
...
+        }
+        if (mutable_descriptor_type->mutableDescriptorTypeListCount > j) {
+            for (uint32_t k = 0; k < mutable_descriptor_type->pMutableDescriptorTypeLists[j].descriptorTypeCount; ++k) {
+                second_types.push_back(mutable_descriptor_type->pMutableDescriptorTypeLists[j].pDescriptorTypes[k]);
+            }
+        } else {
+            second_types = all_descriptor_types;
+        }
+
+        if (first_types.size() != second_types.size()) {
```
Comment (Tony-LunarG) [inline layers/parameter_validation_utils.cpp:4721]: Completely disjoint , non overlapping lists of different sizes will trigger a false positive here.

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp [hunk 2]
```diff
@@ -9780,6 +9795,110 @@ TEST_F(VkLayerTest, InvalidCreateDescriptorPoolFlags) {
...
+
+    lists[1].pDescriptorTypes = first_types;
+    mdtci.mutableDescriptorTypeListCount = 1;
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkDescriptorPoolCreateInfo-pPoolSizes-04787");
+    vk::CreateDescriptorPool(m_device->device(), &ds_pool_ci, NULL, &bad_pool);
+    m_errorMonitor->VerifyFound();
+
+    mdtci.mutableDescriptorTypeListCount = 2;
+    m_errorMonitor->ExpectSuccess();
+    vk::CreateDescriptorPool(m_device->device(), &ds_pool_ci, NULL, &bad_pool);
```
Comment (Tony-LunarG) [inline tests/vklayertests_descriptor_renderpass_framebuffer.cpp:9898]: This call succeeds and creates a pool that needs to be destroyed

**Review by Tony-LunarG — COMMENTED** [review]
A couple of comments

File: layers/parameter_validation_utils.cpp (hunk 1 above)
Comment (ziga-lunarg) [reply layers/parameter_validation_utils.cpp]: I got this pretty wrong, good catch

**Pull request merged**